components/application/widgets/taskEditor/TaskEditor.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * @param
+ * formType: ["add", "edit"]
+ * taskData: {
+ *      selectedDate: "2023-12-04",
+ *      priority: 4,
+ *      taskName: "task name",
+ *      taskContent: "task content",
+ *      tags: ["tag1", "tag2"]
+ * }
+ * tagList: All user tags. Use state in parent component to update the list in case of create a new tag
+ * cancelCallBack: A function to be executed after the cancel btn is clicked
+ * submitCallBack: A function to be executed after the add/save btn is clicked
+ * @usage
+ * <TaskEditor formType="edit" taskData={...} cancelCallBack={callBackFn} submitCallBack = {callBackFn}/>
+ * @todo
+ * 1. How to submit the form
+ * 2. Show date when type='edit'
+ * ...
+ */
+
+import { useRef, useState } from "react";
+import styles from "../../../../styles/scss/components/application/widgets/taskEditor.module.scss";
+import Scheduler, { convertDate } from "../Scheduler";
+import PriorityPicker from "../PriorityPicker";
+import TaskNameInput from "./TaskNameInput";
+import TaskTagCheckList from "./TaskTagCheckList";
+
+function TaskEditor({
+  formType,
+  taskData,
+  tagList,
+  cancelCallBack,
+  submitCallBack,
+}) {
+  // Default values
+  if (formType === null) {
+    formType = "add";
+  }
+  if (taskData === null) {
+    taskData = {
+      id: "",
+      selectedDate: "",
+      priority: 4,
+      taskName: "",
+      taskContent: "",
+      tags: ["Reading1", "Daily"],
+    };
+  }
+  if (tagList === null) {
+    tagList = ["Reading1", "Reading2", "Daily", "Weekly"];
+  }
+  if (cancelCallBack === null) {
+    cancelCallBack = () => {};
+  }
+  if (submitCallBack === null) {
+    submitCallBack = () => {};
+  }
+
+  const taskNameInputRef = useRef(null);
+
+  /** record the editing task */
+  let newTaskData = useRef({
+    id: taskData.id === null ? "" : taskData.id,
```

Comment:
When I Click addTask button, it show this Error
![image](https://github.com/zli213/SynergyStudy/assets/119087910/2043d91f-b169-4562-9791-18b331195883)

Reply:
fixed